From b82f954e71be45aef152bbf8925a3070b7e4c69a Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date: Fri, 5 Apr 2013 02:13:31 +0300
Subject: [PATCH 1/5] Generate config.h after looking for dependencies.

At least lib/config.h.cmake's ICONV_SECOND_ARGUMENT_IS_CONST depends
on FindIconv.cmake having been called before, otherwise it is always
undefined.

This fixes a regression introduced in 61a7ffe.

CCMAIL: cgiboudeaux@gmx.com
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd95c92..b2d959c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,9 +46,6 @@ endif()
 # check for visibility support
 macro_check_gcc_visibility(__STRIGI_HAVE_GCC_VISIBILITY)
 
-# Generate include/strigi/strigiconfig.h and lib/config.h
-include(ConfigureChecks.cmake)
-
 # check for required packages
 find_package(ZLIB)
 set_package_properties(ZLIB PROPERTIES
@@ -77,6 +74,9 @@ set_package_properties(Threads PROPERTIES
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
+# Generate include/strigi/strigiconfig.h and lib/config.h
+include(ConfigureChecks.cmake)
+
 ##### building and testing #####
 # every directory needs the headers that will be installed
 include_directories(${libstreams_SOURCE_DIR}/include
-- 
1.8.4.2

